common/src/java/com/zimbra/common/util/SameSiteAttribute.java

```diff
@@ -0,0 +1,37 @@
+package com.zimbra.common.util;
+
+import java.util.Collection;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @since 4.0.9
+ * @author keshavbhatt
+ *     <p>Utility class to set SameSite attribute of the Set-Cookie HTTP response header. SameSite
+ *     attribute allows us to declare if the cookie should be restricted to a first-party or
+ *     same-site context
+ */
+class SameSiteAttribute {
+
+  private static final String SET_COOKIE_HEADER = "Set-Cookie";
+
+  private SameSiteAttribute() {}
+
+  /**
+   * @param response the response object containing the Set-Cookies header
+   * @param sameSiteValue value to be set for SameSite attribute
+   */
+  public static void addSameSiteAttribute(HttpServletResponse response, String sameSiteValue) {
+    Collection<String> headers = response.getHeaders(SET_COOKIE_HEADER);
+    boolean firstHeader = true;
+    for (String header : headers) {
+      if (firstHeader) {
+        response.setHeader(
+            SET_COOKIE_HEADER, String.format("%s; %s", header, "SameSite=" + sameSiteValue));
+        firstHeader = false;
+        continue;
+      }
+      response.addHeader(
+          SET_COOKIE_HEADER, String.format("%s; %s", header, "SameSite=" + sameSiteValue));
+    }
```

Comment:
I checked HttpServletResponse interface and it seems that the **getHeaders()** method returns a list of header values (not headers). Also the addHeader() method will add values to a header, so this is what I would do:
```suggestion
    if (!(Objects.isNull(response.getHeader(SET_COOKIE_HEADER)))) {
      response.addHeader(SET_COOKIE_HEADER, "SameSite=" + sameSiteValue);
    }
```
The result of my suggestion should be the value "SameSite=something" added to Set-Cookie header.
**I fear the solution you proposed may add a "SameSite=Lax" to each value of the header** (e.g.: Set-Cookie: id=1; SameSite=Lax; Max-Value=b; SameSite=Lax).
**Unfortunately the Unit Test is not able to show and test if this is the case, because MockHttpServletResponse does not implement all methods the right way.** If possible I would check with a "real" HttpServletResponse (IT)

Reply:
I will test it to see if it works.

Reply:
It doesn't work. 
> I fear the solution you proposed may add a "SameSite=Lax" to each value of the header 
It won't :) we have the checks are in place. 
The provided solution is tested and fulfills the requirements already.

Reply:
> It won't :) we have the checks are in place. The provided solution is tested and fulfills the requirements already.
There is only a test with a MockHttpServletResponse.
To fulfill the requirements we need a test with a real HttpResponse and a Set-Cookie header with multiple values

Reply:
We checked together that the implementation result is as expected!
It will just be a plus if we could test SameSiteAttribute class against jetty Response class! Thanks :)